graph/options/options.go

```diff
@@ -61,18 +62,17 @@ func NewOptions(r *http.Request) Options {
 	namespaces := params.Get("namespaces") // csl of namespaces. Overrides namespace path param if set
 	vendor := params.Get("vendor")
 
+        accessibleNamespaces := getAccessibleNamespaces()
+
 	var namespaceNames []string
 	fetchNamespaces := namespaces == NamespaceAll || (namespaces == "" && (namespace == NamespaceAll))
 	if fetchNamespaces {
-		namespaces, err := models.GetNamespaces()
-		checkError(err)
-
-		for _, namespace := range namespaces {
+		for namespace, _ := range accessibleNamespaces {
 			// The istio-system namespace is only shown when explicitly requested. Note that the
 			// 'includeIstio' option doesn't apply here, that option affects what is done in
 			// non-istio-system namespaces.
-			if namespace.Name != NamespaceIstioSystem {
-				namespaceNames = append(namespaceNames, namespace.Name)
+			if namespace != NamespaceIstioSystem {
+				namespaceNames = append(namespaceNames, namespace)
 			}
 		}
 	} else if namespaces != "" {
```

Comment:
After the namespaceNames array is set you should ensure that all of the names are in the accessibleNamespaces set.  If not fail fast like:

`checkError(errors.New(fmt.Sprintf("Requested namespace [%s] is not accessible", namespaceName))`

By validating the requested namespaces here the graph generation code does not have to worry about it at all.